engine/framework/master.go

```diff
@@ -93,6 +94,9 @@ type MasterImpl interface {
 
 	// CloseImpl is called when the master is being closed
 	CloseImpl(ctx context.Context) error
+
+	// StopImpl is called when the master is being canceled
```

Comment:
I think the comment for `CloseImpl` and `StopImpl` is inaccurate.
* `CloseImpl` is called when the job is not the terminated state;
* `StopImpl` is called when the job is  the terminated state;
terminated state is finished/failed/canceled.

Reply:
I think `StopImpl` is called when job is failed or canceled, `finished` job should use `CloseImpl` logic.

Reply:
Have we agreed that `CloseImpl` is called `when the job is not the terminated state`?